packages/core/upload/admin/src/components/TableList/TableRows.js

```diff
@@ -16,42 +11,59 @@ import Pencil from '@strapi/icons/Pencil';
 import Eye from '@strapi/icons/Eye';
 
 import { CellContent } from './CellContent';
+import { isSelectable } from './utils/isSelectable';
 import { AssetDefinition, FolderDefinition, tableHeaders as cells } from '../../constants';
-import { getFolderURL, getTrad } from '../../utils';
+import { getTrad, toSingularTypes } from '../../utils';
 
 export const TableRows = ({
+  allowedTypes,
   canUpdate,
+  isFolderSelectionAllowed,
+  onChangeFolder,
   onEditAsset,
   onEditFolder,
   onSelectOne,
   rows,
   selected,
 }) => {
   const { formatMessage } = useIntl();
-  const { pathname } = useLocation();
-  const [{ query }] = useQueryParams();
-  const { push } = useHistory();
 
-  const handleRowClickFn = (element, elementType) => {
+  const handleRowClickFn = (element, elementType, id) => {
     if (elementType === 'asset') {
       onEditAsset(element);
     } else {
-      push(getFolderURL(pathname, query, element));
+      onChangeFolder(id);
     }
   };
 
+  const singularTypes = toSingularTypes(allowedTypes);
+
   return (
     <Tbody>
       {rows.map((element) => {
-        const { alternativeText, id, name, ext, url, mime, formats, type: elementType } = element;
+        const {
+          alternativeText,
+          id,
+          name,
+          ext,
+          url,
+          mime,
+          folderURL,
+          formats,
+          type: elementType,
+        } = element;
+
+        const fileType = mime?.split('/')?.[0];
```

Comment:
this feels very flakey, like it could break, would it be better to ensure mime by default is an empty string above? so then you're just doing:

```js
const [fileType] = mime.split('/')
```

which will resolve to an empty string.